cognite/client/utils/_identifier.py

```diff
@@ -226,3 +239,31 @@ class SingletonIdentifierSequence(IdentifierSequenceCore[Identifier]):
 
 class DataModelingIdentifierSequence(IdentifierSequenceCore[DataModelingIdentifier]):
     ...
+
+
+class UserIdentifierSequence(IdentifierSequenceCore[UserIdentifier]):
+    @classmethod
+    def load(cls, user_identifiers: str | Sequence[str]) -> UserIdentifierSequence:
+        if isinstance(user_identifiers, str):
+            return cls(identifiers=[UserIdentifier(user_identifiers)], is_singleton=True)
+
+        elif isinstance(user_identifiers, Sequence):
+            return cls(identifiers=list(map(UserIdentifier, map(str, user_identifiers))), is_singleton=False)
+
+        raise TypeError(f"user_identifiers must be of type str or Sequence[str]. Found {type(user_identifiers)}")
+
+    def chunked(self, chunk_size: int) -> list[UserIdentifierSequence]:  # type: ignore [override]
+        return [
+            UserIdentifierSequence(chunk, is_singleton=self.is_singleton())
+            for chunk in split_into_chunks(self._identifiers, chunk_size)
+        ]
```

Comment:
If we change the parent implementation to 
```python
    def chunked(self, chunk_size: int) -> list[Self]:
        return [
            type(self)(chunk, is_singleton=self.is_singleton())
            for chunk in split_into_chunks(self._identifiers, chunk_size)
        ]
```
I don't think this will be necessary.

Reply:
This was my first version actually, but our (old) version of `mypy` doesn't work with `Self` (it seems):
```
...error: Variable "typing_extensions.Self" is not valid as a type  [valid-type]
        def chunked(self, chunk_size: int) -> list[Self]:
```

Reply:
Made `mypy` happy by changing to:
```py
    def chunked(self: T_IdentifierSequenceCore, chunk_size: int) -> list[T_IdentifierSequenceCore]:
        (...)

T_IdentifierSequenceCore = TypeVar("T_IdentifierSequenceCore", bound=IdentifierSequenceCore)